Library/Homebrew/cmd/desc.rb

```diff
@@ -18,8 +18,8 @@ def desc_args
     Homebrew::CLI::Parser.new do
       description <<~EOS
         Display <formula>'s name and one-line description.
-        Formula descriptions are cached; the cache is created on the
-        first search, making that search slower than subsequent ones.
+        Descriptions are cached if HOMEBREW_EXEC_ALL is set.
+        The cache is created on the first search, making that search slower than subsequent ones.
```

Comment:
To be sure, will `--search` still work without `HOMEBREW_EXEC_ALL`? Doesn't `--search` need to read all formulae?

Reply:
@carlocab Yes. It uses formula filenames, not their contents.